source/extensions/filters/http/ip_tagging/ip_tagging_filter.h

```diff
@@ -39,6 +39,8 @@ class IpTaggingFilterConfig {
   FilterRequestType requestType() const { return request_type_; }
   const Network::LcTrie::LcTrie<std::string>& trie() const { return *trie_; }
 
+  const Http::LowerCaseString& ip_tag_header() const { return header_; }
```

Comment:
Please add a comment here, stating that empty string implies that no `ip_tag_header` is set.

I do wonder if it just makes more sense to return the correct `ip_tag_header` here (either the default or the alternative). On the one hand I think it may make the code a bit more readable, but then the calls to the headers.append will need to be a bit different.

Reply:
Yeah I added the conditional in `decodeHeaders` as first, it was easier and second the decision on which header to use is made when we are appending the tags but I'm open to returning the correct header from `ip_tag_header()`. I will have to see how involved that change is but let me know if that's what you prefer.